.git-hooks/pre-commit

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env node
+
+const { execSync } = require("child_process");
+const { existsSync, promises: fsPromises } = require("fs");
+const { join, resolve } = require("path");
+
+const excludedDirs = ["node_modules"];
+
+const getAllJsFiles = async (dir) => {
+  const getFilesRecursively = async (currentDir) => {
+    const files = await fsPromises.readdir(currentDir);
+    let jsFiles = [];
+
+    for (const file of files) {
+      const filePath = join(currentDir, file);
+      const fileStat = await fsPromises.stat(filePath);
+
+      if (fileStat.isDirectory() && !excludedDirs.includes(file)) {
+        jsFiles.push(...(await getFilesRecursively(filePath)));
+      } else if (file.endsWith(".js")) {
+        jsFiles.push(filePath);
+      }
+    }
+
+    return jsFiles;
+  };
+
+  try {
+    return await getFilesRecursively(dir);
+  } catch (error) {
+    console.error("Error occurred while getting JS files:", error.message);
+    throw error;
+  }
+};
+
+const runPrettierOnAllJsFiles = async () => {
+  try {
+    const projectDir = resolve(process.cwd());
+    const allJsFiles = await getAllJsFiles(projectDir);
+
+    console.log("Code formatting...");
+
+    if (allJsFiles.length > 0) {
+      let prettierCmd = "npx prettier";
+      const localPrettierPath = resolve(projectDir, "node_modules/.bin/prettier");
+      if (existsSync(localPrettierPath)) {
+        prettierCmd = localPrettierPath;
+      }
+
+      prettierCmd += ` --config ${resolve(projectDir, ".prettierrc.json")} --write ${allJsFiles.join(" ")}`;
+
+      execSync(prettierCmd, { stdio: "inherit" });
+
+      console.log("Prettier formatting applied to all JS files.");
+
+      const gitAddCmd = `git add ${allJsFiles.join(" ")}`;
```

Comment:
Is this going to add (stage) everything even if I try to add one file?

Reply:
@lugenx you can add just one file. However, during git commit execution, all .js files will be formatted with prettier, to maintain consistent styling even if local formatting tools/settings are used by some devs. 
Then formatted files are staged, ensuring accurate tracking by Git version control, which might not always catch formatting changes as effectively as content changes.
If your added file is the only one with style or content differences, only that one gets pushed

Reply:
@lugenx seeing that this PR itself formatted all files to fit the prettier configuration settings, any updated forks making commits after this merge will be formatted the same. So any files that are updated in future PRs should be the only files pushed.